vcs/git.go

```diff
@@ -151,6 +152,51 @@ func (g *GitDriver) SpecialFiles() []string {
 	}
 }
 
+func (g *GitDriver) AutoGeneratedFiles(dir string) []string {
+	var files []string
+
+	filesCmd := exec.Command("git", "ls-files", "-z");
+	filesCmd.Dir = dir
+	pipe, err := filesCmd.StdoutPipe();
+
+	if err != nil {
+		log.Printf("Error occured when running git ls-files in %s: %s.", dir, err)
+		return files
+	}
+
+	if err := filesCmd.Start(); err != nil {
+		log.Printf("Error occured when running git ls-files in %s: %s.", dir, err)
+		return files
+	}
+
+	attributesCmd := exec.Command("git", "check-attr", "--stdin", "-z", autoGeneratedAttribute);
+	attributesCmd.Dir = dir
+	attributesCmd.Stdin = pipe
+
+	out, err := attributesCmd.Output()
+
+	if err != nil {
+		log.Printf("Error occured when running git check-attr in %s: %s.", dir, err)
+		return files
+	}
+
+	// Split by NUL and we expect the format: <path> NUL <attribute> NUL <info> NUL
+	tokens := bytes.Split(out, []byte{0})
+
+	for i := 2; i < len(tokens); i+=3 {
+		if string(tokens[i]) == "true" && string(tokens[i-1]) == autoGeneratedAttribute {
+			files = append(files, string(tokens[i-2]))
+		}
+	}
+
+	if err := filesCmd.Wait(); err != nil {
```

Comment:
Followed the pattern in the example in https://pkg.go.dev/os/exec#Cmd.StdoutPipe so hopefully this is right